keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -80,13 +84,17 @@ This includes defining conditions (Waiting, Running, Progressing, Available) for
 ## Motivation
 
 Today only deployment controller has a [status](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status) to fully reflect the state during its lifecycle.
-This enhancement extends the scope of those and other conditions to other controllers (DaemonSet, Job, ReplicaSet & ReplicationController, StatefulSet).
+This enhancement extends the scope of those and other conditions to other controllers (DaemonSet, Job, ReplicaSet & ReplicationController, StatefulSet, Deployment) where applicable.
+
+These conditions can be used by high-level controllers, tools, and end users in an effort to hide specifics of each workload
```

Comment:
We have different workload APIs because they behave very different, so I don't think we can necessarily generalize conditions.

In particular, I think we should give up on generalizing "Available" or "Operational", as you are suggesting.

Reply:
Some of them behave different, but we try to suggest only the workloads which have common characteristics.

- Deployment, StatefulSet, DaemonSet: all of them have a rollout dynamic in common, and would benefit from `Progressing` condition to know the state of a rollout 
- ReplicaSet, Deployment, StatefulSet, DaemonSet: the responsibility of these controllers is that they should heal the pods and make sure the user application has enough pods for it to be healthy: I think `Opeartional` fits this use case